src/core/basic_client.rs

```diff
@@ -672,6 +677,118 @@ impl BasicClient {
+    ///
+    /// # Errors
+    ///
+    /// If the implicit client provider is `ProviderID::Core`, a client error
+    /// of `InvalidProvider` type is returned.
+    ///
+    /// If the implicit client provider has not been set, a client error of
+    /// `NoProvider` type is returned.
+    ///
+    /// See the operation-specific response codes returned by the service
+    /// [here](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_aead_encrypt.html#specific-response-status-codes).
+    pub fn psa_aead_encrypt(
+        &self,
+        key_name: String,
+        encrypt_alg: Aead,
+        nonce: &[u8],
+        additional_data: &[u8],
+        plaintext: &[u8],
+    ) -> Result<Vec<u8>> {
+        let crypto_provider = self.can_provide_crypto()?;
+
+        let op = PsaAeadEncrypt {
+            key_name,
+            alg: encrypt_alg,
+            nonce: nonce.to_vec().into(),
+            additional_data: additional_data.to_vec().into(),
+            plaintext: plaintext.to_vec().into(),
+        };
+
+        let encrypt_res = self.op_client.process_operation(
+            NativeOperation::PsaAeadEncrypt(op),
+            crypto_provider,
+            &self.auth_data,
+        )?;
+
+        if let NativeResult::PsaAeadEncrypt(res) = encrypt_res {
+            Ok(res.ciphertext.to_vec())
+        } else {
+            // Should really not be reached given the checks we do, but it's not impossible if some
+            // changes happen in the interface
+            Err(Error::Client(ClientErrorKind::InvalidServiceResponseType))
+        }
+    }
+
+    /// **[Cryptographic Operation]** Decrypt and authenticate a short message.
+    ///
+    /// The key intended for decrypting **must** have its `decrypt` flag set
+    /// to `true` in its [key policy](https://docs.rs/parsec-interface/*/parsec_interface/operations/psa_key_attributes/struct.Policy.html).
+    ///
+    /// The decryption will be performed with the algorithm defined in `alg`,
+    /// but only after checking that the key policy and type conform with it.
+    ///
+    /// `nonce` must be appropriate for the selected `alg`.
+    ///
+    /// For algorithms where the encrypted data and the authentication tag are defined as separate inputs,
+    /// `ciphertext` must contain the encrypted data followed by the authentication data.
+    ///
+    /// # Errors
+    ///
+    /// If the implicit client provider is `ProviderID::Core`, a client error
+    /// of `InvalidProvider` type is returned.
+    ///
+    /// If the implicit client provider has not been set, a client error of
+    /// `NoProvider` type is returned.
+    ///
+    /// See the operation-specific response codes returned by the service
+    /// [here](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_aead_decrypt.html#specific-response-status-codes).
+    pub fn psa_aead_decrypt(
+        &self,
+        key_name: String,
+        encrypt_alg: Aead,
```

Comment:
Should this be `decrypt_alg`? I see the spec left it as `alg`

Reply:
Wasn't sure. They are grouped as 'encryption algorithms', which is why I left it as `encrypt_alg` but I suppose that could cause confusion.

Reply:
It might be why they left it at `alg` in the first place. At the same time, you would preferably decrypt with the same algorithm as the one you encrypted with 😄

Reply:
Leave it as it is, to be consistent with the asymmetric ones